src/mongo_logger.ts

```diff
@@ -421,13 +453,100 @@ export function stringifyWithMaxLen(
 ): string {
   let strToTruncate = '';
 
+  let currentLength = 0;
+  const maxDocumentLengthEnsurer = function maxDocumentLengthEnsurer(key: string, value: any) {
+    if (currentLength >= maxDocumentLength) {
+      return undefined;
+    }
+    // Account for root document
+    if (key === '') {
+      // Account for starting brace
+      currentLength += 1;
+      return value;
+    }
+
+    // +4 accounts for 2 quotation marks, colon and comma after value
+    currentLength += key.length + 4;
+
+    if (value == null) return value;
+
+    switch (typeof value) {
+      case 'string':
+        // +2 accounts for quotes
+        currentLength += value.length + 2;
+        break;
+      case 'number':
+      case 'bigint':
+        currentLength += String(value).length;
+        break;
+      case 'boolean':
+        currentLength += value ? 4 : 5;
+        break;
+      case 'object':
+        if ('buffer' in value && isUint8Array(value.buffer)) {
+          // Handle binData
+          currentLength += (value.buffer.byteLength + value.buffer.byteLength * 0.5) | 0;
```

Comment:
What's this value supposed to be? we handle Binary lower down

if value is a Uint8Array then value.buffer is an ArrayBuffer

Reply:
This was the implementation that was there before 😅, I can go ahead and try and figure out either a more accurate formula for this if it's not slow or remove it

Reply:
No I mean what type is this value in practice, sorry the previous implementation was full of shots in the dark. 

Folks can use Uint8Array directly as binData input but this condition seems like it's halfway to detecting Binary not Uint8Array